From 9f7a75333af879f85938bebcecde313c7a0d3519 Mon Sep 17 00:00:00 2001
From: Peter Rifel <pgrifel@gmail.com>
Date: Wed, 18 Jun 2025 05:41:42 -0700
Subject: [PATCH] Remove --register-schedulable kubelet flag in 1.34+

The --register-schedulable kubelet flag was removed in Kubernetes 1.34.
This patch conditionally sets the flag only for Kubernetes versions < 1.34.

Backported from PR #17443 for v1.32.1 compatibility.
---
 pkg/model/components/kubelet.go | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/pkg/model/components/kubelet.go b/pkg/model/components/kubelet.go
index ce0c0823ad..37246ec7cd 100644
--- a/pkg/model/components/kubelet.go
+++ b/pkg/model/components/kubelet.go
@@ -206,7 +206,9 @@ func (b *KubeletOptionsBuilder) configureKubelet(cluster *kops.Cluster, kubelet
 		kubelet.ShutdownGracePeriodCriticalPods = &metav1.Duration{Duration: 0}
 	}
 
-	kubelet.RegisterSchedulable = fi.PtrTo(true)
+	if kubernetesVersion.IsLT("1.34") {
+		kubelet.RegisterSchedulable = fi.PtrTo(true)
+	}
 
 	return nil
 }
-- 
2.39.3 (Apple Git-146)
